### DISCUSSION TITLE
Invalidate and list all active tokens

### DISCUSSION THREAD

**Discussion by yepes** [pocketbase/pocketbase#577] [Q&A] [answered]
Hi,

Is it possible to list / invalidate all tokens (and only by a user) received by authViaEmail?

**Comment by ganigeorgiev** [comment, marked as answer]
**You can't list previously issued tokens because they aren't stored in the database.** PocketBase REST API is completely stateless.

You can invalidate all tokens by type (auth, forgotten password, etc.) from the Admin UI by going to "Settings > Token options" and click on the "Invalidate all previously issued tokens" available under each token type input and then Save the form.

Currently there is no option from the UI (or the REST API) to directly invalidate the tokens for a single user. This happens automatically for example on  user password change. You can trigger it programmatically only if you use PocketBase as framework and call `RefreshTokenKey()` on the `*models.User` instance and save it.

**Reply by ganigeorgiev** [reply]
_btw. you could find a short expanation how PocketBase tokens are constructed here - https://github.com/pocketbase/pocketbase/issues/144#issuecomment-1186144841_

**Reply by yepes** [reply]
Thanks!
